### PR TITLE
Add MobileStepper component

### PR DESCRIPTION
**Pull request by daileytj**
Fixes PXBLUE-1590.

#### Changes proposed in this Pull Request:
- Add MobileStepper component
- Add example MobileStepper to showcase
- Add MobileStepper story
- Add unit tests for MobileStepper component

### CODE REVIEW

File: components/src/core/mobile-stepper/mobile-stepper.test.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { MobileStepper } from '.';
+import TestRenderer, { ReactTestInstance } from 'react-test-renderer';
+import { ProgressBar } from 'react-native-paper';
+import { View } from 'react-native';
+
+describe('MobileStepper', () => {
+    it('should render the correct number of steps', () => {
```
Comment (joebochill) [inline components/src/core/mobile-stepper/mobile-stepper.test.tsx:8]: May want a few edge case tests here...what happens if they set steps={-5} or steps={0} or activeStep={null}

Comment (jeffvg) [reply components/src/core/mobile-stepper/mobile-stepper.test.tsx:8]: are we able to set steps based on the active steps with the knobs? If steps are set to 4, active to be the same...
image

Comment (joebochill) [reply components/src/core/mobile-stepper/mobile-stepper.test.tsx]: We can bound it in the component so that active step is never higher than the number of steps. Dynamically changing the knobs in the storybook is more challenging though...Tom tried it and ran into some issues getting it to work, so we backed down on that for now.

File: components/src/core/mobile-stepper/mobile-stepper.tsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+                        {adjustedActiveStep + 1} / {adjustedSteps}
+                    </Text>
+                )}
+
+                {variant === 'progress' && (
+                    <View style={{ flex: 1 }}>
+                        <ProgressBar
+                            style={[defaultStyles.progressBar, styles.progressBar]}
+                            progress={adjustedActiveStep / (adjustedSteps - 1)}
+                            color={theme.colors.primary}
```
Comment (joebochill) [inline components/src/core/mobile-stepper/mobile-stepper.tsx:119]: can we use the activeColor on the progressbar too?

**Review by joebochill — APPROVED** [review]
Looks good. If you want to add a test to check the input color props, it might be nice. But if you're tired of looking at this, you can put in a placeholder comment and make it a new enhancement (with the testing label).

**Review by joebochill — APPROVED** [review]
